src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -322,6 +323,7 @@ public bool IsSolutionDPLEnabled
                 {
                     await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+                    EnsureCacheInitialized();
```

Comment:
I don't think full cache initialization is needed here. This operation can be extremely heavy.